src/util/sharing_map.h

```diff
@@ -317,6 +354,11 @@ class sharing_mapt
   void gather_all(const baset &n, const unsigned depth, delta_viewt &delta_view)
     const;
 
+  unsigned count_unmarked_nodes(
+    bool leafs_only,
+    std::set<void *> &marked,
```

Comment:
Phew, what is this meant to be? What can be done using a set of `void` pointers?

Reply:
The set `marked` contains pointers to the nodes with `use_count >= 2` that have been visited already during the traversal. It's used to avoid exploring shared subtrees more than once.

The sharing map has two different types of nodes that don't have a common base class, so I used `void` pointers here to avoid having to keep two separate sets `marked` for both node types.